time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalTimeArbitrary.java

```diff
@@ -19,72 +20,42 @@
 
 	private int hourMin = 0;
 	private int hourMax = 23;
-	private int hourPeriod;
 
 	private int minuteMin = 0;
 	private int minuteMax = 59;
-	private int minutePeriod;
 
 	private int secondMin = 0;
 	private int secondMax = 59;
-	private int secondPeriod;
 
-	private int millisecondMin = 0;
-	private int millisecondMax = 999;
-	private int millisecondPeriod;
-
-	private int microsecondMin = 0;
-	private int microsecondMax = 999;
-	private int microsecondPeriod;
-
-	private int nanosecondMin = 0;
-	private int nanosecondMax = 999;
-	private int nanosecondPeriod;
-
-	private long helperMinuteSecondMilliMicroNano;
-	private long helperSecondMilliMicroNano;
-	private long helperMilliMicroNano;
-	private long helperMicroNano;
+	private ChronoUnit precision = NANOS;
 
 	@Override
 	protected Arbitrary<LocalTime> arbitrary() {
 
-		LocalTime effectiveMin = calculateEffectiveMin();
+		LocalTime effectiveMin;
+		try {
+			effectiveMin = calculateEffectiveMin();
+		} catch (NoValuesCanBeGeneratedException e) {
+			//TODO: Other Exception??
+			throw new TooManyFilterMissesException(e.getMessage());
+		}
 		LocalTime effectiveMax = calculateEffectiveMax();
+		if (effectiveMax.isBefore(effectiveMin)) {
+			//TODO: Other Exception??
+			throw new TooManyFilterMissesException("The maximum time is too soon after the minimum time.");
+		}
```

Comment:
Is this a good exception or should I throw another exception?

Reply:
How can I say that no values can be generated with this configuration at the runtime instead?

Reply:
In the end I guess an IllegalArgumentException is the right one, cause this shows that arbitrary has been wrongly configured, right?